java/arcs/core/data/Ttl.kt

```diff
@@ -24,15 +26,33 @@ sealed class Ttl(count: Int, val isInfinite: Boolean = false) {
             "must be either positive count on infinite, " +
                 "but got count=$count and isInfinite=$isInfinite"
         }
+        if (!isInfinite) {
+            requireNotNull(time) {
+                "time cannot be null for non infinite ttl"
+            }
+        }
     }
 
     override fun equals(other: Any?): Boolean =
         other is Ttl && minutes == other.minutes
 
     override fun hashCode(): Int = minutes.hashCode()
 
-    data class Minutes(val count: Int) : Ttl(count)
-    data class Hours(val count: Int) : Ttl(count)
-    data class Days(val count: Int) : Ttl(count)
-    object Infinite : Ttl(-1, true)
+    @Suppress("")
+    fun calculateExpiration(): Long {
```

Comment:
To avoid needing the globally-mutable `time` variable in the companion object, maybe this method should accept a `Time`.

Reply:
i totally agree. but the Handle classes are also not platform dependent, where could they get the `Time` from?

Reply:
it could be a required param?

Reply:
moved `Time` to a Handle val and assed as parameter here. WDYT?